screeps-game-api/src/game.rs

```diff
@@ -170,12 +172,185 @@ pub mod map {
+    pub struct Transaction {
+        transaction_id: String,
+        time: u32,
+        sender: Player,
+        recipient: Player,
+        resource_type : String,
+        amount: u32,
+        from: String,
+        to: String,
+        description: String,
+        order: Option<TransactionOrder>,
+    }
+    js_deserializable!(Transaction);
+    
+    #[derive(Deserialize, Debug)]
+    #[serde(rename_all = "camelCase")]
+    pub struct Order {
+        id: String,
+        created: u32,
+        #[serde(rename = "type")]
+        order_type: String,
+        resource_type: String,
+        room_name: String,
+        amount: u32,
+        remaining_amount: u32,
+        price: f64
+    }
+    js_deserializable!(Order);
+
+    #[derive(Deserialize, Debug)]
+    #[serde(rename_all = "camelCase")]
+    pub struct MyOrder {
+        id: String,
+        created: u32,
+        active: bool,
+        #[serde(rename = "type")]
+        order_type: String,
+        resource_type: String,
+        room_name: String,
+        amount: u32,
+        remaining_amount: u32,
+        total_amount: u32,
+        price: f64
+    }
+    js_deserializable!(MyOrder);
+
+    pub fn credits() -> u32 {
+        js_unwrap!(Game.market.credits)
+    }
+
+    pub fn incoming_transactions() -> Vec<Transaction>{
+        let arr_transaction_value = js!{
+            return Game.market.incomingTransactions;
+        };
+        arr_transaction_value.try_into().unwrap()
+    }
+
+    pub fn outgoing_transactions() -> Vec<Transaction>{
+        let arr_transaction_value = js!{
+            return Game.market.outgoingTransactions;
+        };
+        arr_transaction_value.try_into().unwrap()
+    }
+
+    pub fn orders() -> HashMap<String, MyOrder> {
+        let order_book_value = js! {
+            return Game.market.orders;
+        };
+        order_book_value.try_into().unwrap()
+    }
+
+    pub fn calc_transaction_cost(amount: u32, room1: &Room, room2: &Room) -> u32 {
+        js_unwrap!(Game.market.calcTransactionCost(@{amount}, @{room1.name()}, @{room2.name()}))
+    }
+
+    pub fn cancel_order(order_id: &str) -> ReturnCode {
+        js_unwrap!(Game.market.cancelOrder(@{order_id}))
+    }
+
+    pub fn change_order_price(order_id: &str, new_price: u32) -> ReturnCode {
+        js_unwrap!(Game.market.changeOrderPrice(@{order_id}, @{new_price}))
+    }
+
+    pub fn create_order(order_type: OrderType, resource_type: ResourceType, 
+                        price: f64, total_amount: u32, room: &Room) -> ReturnCode {
+        js_unwrap!{
+            Game.market.createOrder(@{order_type.as_string()},
+                                    __resource_type_num_to_str(@{resource_type as i32}),
+                                    @{price},
+                                    @{total_amount},
+                                    @{room.name()})
+        }
+    }
+
+    pub fn deal(order_id: &str, amount: u32, target_room: Room) -> ReturnCode {
```

Comment:
Could this take `&Room` instead?

I'd almost want to have it taking a room name `&str` so that someone could execute deals without grabbing a reference to their room, but I don't feel too strongly about that.

Reply:
I do, it was a mistake on my part. Fixed!